packages/block-editor/src/components/block-edit/index.js

```diff
@@ -20,11 +20,19 @@ import { BlockEditContextProvider, useBlockEditContext } from './context';
 export { useBlockEditContext };
 
 export default function BlockEdit( props ) {
-	const { name, isSelected, clientId, __unstableLayoutClassNames } = props;
+	const {
+		name,
+		isSelected,
+		clientId,
+		attributes = {},
+		__unstableLayoutClassNames,
+	} = props;
+	const { layout = null } = attributes;
```

Comment:
This might be super edge-casey, but I was wondering if we should check whether or not the block has opted-in to the layout support before passing down its `layout` attribute in the context? Because this will apply to all blocks, if a 3rd party block happens to use an attribute also named `layout`, then it will be passed down here, too, without a check for the layout block support.

Reply:
Good idea, let's do that!